### PR TITLE
Short-circuit common case of comparing a calendar or time zone to itself

### PR DESCRIPTION
**Pull request by ptomato**
See point 2 of #1425.

### CODE REVIEW

File: polyfill/lib/ecmascript.mjs [hunk 1]
```diff
@@ -1844,11 +1844,13 @@ export const ES = ObjectAssign({}, ES2020, {
     return cal1 < cal2 ? -1 : cal1 > cal2 ? 1 : 0;
   },
   CalendarEquals: (one, two) => {
+    if (one === two) return true;
```
Comment (cjtenny) [inline polyfill/lib/ecmascript.mjs:1847]: too complex, no comments, how can anyone even understand what these added lines do

**Review by cjtenny — APPROVED** [review]
lgtm :)

File: spec/calendar.html [hunk 2]
```diff
@@ -329,6 +329,7 @@ <h1>FormatCalendarAnnotation ( _id_, _showCalendar_ )</h1>
     <emu-clause id="sec-temporal-calendarequals" aoid="CalendarEquals">
       <h1>CalendarEquals ( _one_, _two_ )</h1>
       <emu-alg>
+        1. If _one_ and _two_ are the same Object value, return *true*.
```
Comment (ljharb) [inline spec/calendar.html:332]: this AO doesn't assert that its arguments are of type `Object` (which should be done in its description prose, which it's currently missing)

File: spec/calendar.html [hunk 3]
```diff
@@ -339,6 +340,7 @@ <h1>CalendarEquals ( _one_, _two_ )</h1>
     <emu-clause id="sec-temporal-consolidatecalendars" aoid="ConsolidateCalendars">
       <h1>ConsolidateCalendars ( _one_, _two_ )</h1>
       <emu-alg>
+        1. If _one_ and _two_ are the same Object value, return _two_.
```
Comment (ljharb) [inline spec/calendar.html:343]: same here

File: spec/timezone.html [hunk 4]
```diff
@@ -640,6 +640,7 @@ <h1>TimeZoneEquals ( _one_, _two_ )</h1>
         calling `toString()` on its arguments are equal.
       </p>
       <emu-alg>
+        1. If _one_ and _two_ are the same Object value, return *true*.
```
Comment (ljharb) [inline spec/timezone.html:643]: same here

**Review by ljharb — CHANGES_REQUESTED** [review]

**Pull request merged**